sdk/core/azure-core/azure/core/tracing/_abstract_span.py

```diff
@@ -42,7 +43,7 @@ class SpanKind(Enum):
     INTERNAL = 6
 
 
-class AbstractSpan(Protocol):
+class AbstractSpan(Protocol, Generic[SpanType]):
```

Comment:
Instead of a generic protocol, could this be a good opportunity to use Self? https://peps.python.org/pep-0673/#use-in-protocols

Reply:
No, the generic type here will be the `Span` implementation by the plugin, for instance `opentelemetry.Span`. It's not a recursive reference to itself.